src/axom/sidre/tests/sidre_mfem_datacollection.cpp

```diff
@@ -774,6 +774,71 @@ TEST(sidre_datacollection, dc_par_reload_gf)
   EXPECT_TRUE(sdc_reader.verifyMeshBlueprint());
 }
 
+TEST(sidre_datacollection, dc_par_reload_gf_ordering)
+{
+  const std::string first_field_name = "test_field_1";
+  const std::string second_field_name = "test_field_2";
+
+  // 3D tet mesh
+  mfem::Mesh mesh(2, 2, 2, mfem::Element::TETRAHEDRON);
+  mfem::ParMesh parmesh(MPI_COMM_WORLD, mesh);
+
+  mfem::H1_FECollection fec(1, mesh.Dimension());
+  mfem::ParFiniteElementSpace first_parfes(&parmesh,
+                                           &fec,
+                                           1,
+                                           mfem::Ordering::byNODES);
+  mfem::ParFiniteElementSpace second_parfes(&parmesh,
+                                            &fec,
+                                            3,
+                                            mfem::Ordering::byVDIM);
+
+  // The mesh must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  bool owns_mesh = true;
+  MFEMSidreDataCollection sdc_writer(testName(), &parmesh, owns_mesh);
+
+  // The mesh and field(s) must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  mfem::ParGridFunction first_gf_write(&first_parfes,
+                                       static_cast<double*>(nullptr));
+  mfem::ParGridFunction second_gf_write(&second_parfes,
+                                        static_cast<double*>(nullptr));
+
+  // Register to allocate storage internally, then write to it
+  sdc_writer.RegisterField(first_field_name, &first_gf_write);
+  sdc_writer.RegisterField(second_field_name, &second_gf_write);
+
+  mfem::ConstantCoefficient three_and_a_half(3.5);
+  first_gf_write.ProjectCoefficient(three_and_a_half);
+
+  mfem::ConstantCoefficient five_and_a_half(5.5);
+  second_gf_write.ProjectCoefficient(five_and_a_half);
+
+  sdc_writer.SetCycle(0);
+  sdc_writer.Save();
+
+  MFEMSidreDataCollection sdc_reader(testName());
```

Comment:
Should this be read into a separate datastore? 
Alternatively, should the datastore be cleared to ensure that reading and writing are happening as expected?

Reply:
In this test (and I believe all the others) the datacollections all own their datastores which are created at the time of construction, so I think there wouldn't be anything to clear.  It would be worth adding a test that roughly captures the Serac use case that motivated these changes (per your comment at the top), though.